kratos/input_output/cad_json_input.h

```diff
@@ -0,0 +1,823 @@
+    }
+
+    ///@}
+    ///@name Read in Brep
+    ///@{
+
+    static void ReadBreps(
+        const Parameters& rParameters,
+        ModelPart& rModelPart,
+        SizeType EchoLevel = 0)
+    {
+        for (IndexType brep_index = 0; brep_index < rParameters.size(); brep_index++)
+        {
+            KRATOS_INFO_IF("ReadBreps", (EchoLevel > 0))
+                << "Reading Brep \"" << GetIdOrName(rParameters[brep_index])
+                << "\" - faces." << std::endl;
+
+            ReadBrepFaces(rParameters[brep_index], rModelPart, EchoLevel);
+        }
+
+        for (IndexType brep_index = 0; brep_index < rParameters.size(); brep_index++)
+        {
+            KRATOS_INFO_IF("ReadBreps", (EchoLevel > 0))
+                << "Reading Brep \"" << GetIdOrName(rParameters[brep_index])
+                << "\" - edges." << std::endl;
+
+            ReadBrepEdges(rParameters[brep_index], rModelPart, EchoLevel);
+        }
+    }
+
+
+    static void ReadBrepFaces(
+        const Parameters& rParameters,
+        ModelPart& rModelPart,
+        SizeType EchoLevel = 0)
+    {
+        if (rParameters.Has("faces"))
+        {
+            ReadBrepSurfaces(rParameters["faces"], rModelPart, EchoLevel);
+        }
+    }
+
+    static void ReadBrepEdges(
+        const Parameters& rParameters,
+        ModelPart& rModelPart,
+        SizeType EchoLevel = 0)
+    {
+        if (rParameters.Has("edges"))
+        {
+            ReadBrepCurveOnSurfaces(rParameters["edges"], rModelPart, EchoLevel);
+        }
+    }
+
+    ///@}
+    ///@name Read in Brep Geometries
+    ///@{
+
+    static void ReadBrepSurfaces(
+        const Parameters& rParameters,
+        ModelPart& rModelPart,
+        SizeType EchoLevel = 0)
+    {
+        KRATOS_ERROR_IF_NOT(rParameters.IsArray())
+            << "\"faces\" section needs to be an array of BrepSurfaces." << std::endl;
+
+        KRATOS_INFO_IF("ReadBrepSurfaces", EchoLevel > 2)
+            << "Reading " << rParameters.size() << " BrepSurfaces..." << std::endl;
+
+        for (IndexType brep_surface_i = 0; brep_surface_i < rParameters.size(); ++brep_surface_i)
+        {
+            ReadBrepSurface(rParameters[brep_surface_i], rModelPart, EchoLevel);
+        }
+    }
+
+    static void ReadBrepSurface(
+        const Parameters& rParameters,
+        ModelPart& rModelPart,
+        SizeType EchoLevel = 0)
+    {
+        KRATOS_INFO_IF("ReadBrepSurface", (EchoLevel > 3))
+            << "Reading BrepSurface \"" << GetIdOrName(rParameters) << "\"" << std::endl;
+
+        KRATOS_ERROR_IF_NOT(HasIdOrName(rParameters))
+            << "Missing 'brep_id' or 'brep_name' in brep face." << std::endl;
+
+        KRATOS_ERROR_IF_NOT(rParameters.Has("surface"))
+            << "Missing 'surface' in brep face." << std::endl;
+
+        auto p_surface = ReadNurbsSurface<3, TNodeType>(
+            rParameters["surface"], rModelPart, EchoLevel);
+
+        bool is_trimmed = true;

```

Comment:
I recommend to check the [ternary operator](https://www.geeksforgeeks.org/conditional-or-ternary-operator-in-c-c/)

In general I like it because this way you can make things `const`

Reply:
In this specific case it didn't optimize a lot as there is also a warning if wrong. Anyways, I changed it because it looks cooler like that ;)